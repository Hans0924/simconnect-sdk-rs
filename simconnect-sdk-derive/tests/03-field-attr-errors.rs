#![allow(unused_variables, dead_code)]
use simconnect_sdk_derive::SimConnectObject;

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData1 {
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData2 {
    #[simconnect]
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData3 {
    #[simconnect()]
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData4 {
    #[simconnect(name = "PLANE LATITUDE")]
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData5 {
    #[simconnect(unit = "degrees")]
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData6 {
    #[simconnect(name = "PLANE LATITUDE", name = "PLANE LATITUDE")]
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData7 {
    #[simconnect(unit = "degrees", unit = "degrees")]
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData8 {
    #[simconnect(name = "PLANE LATITUDE", unit = "degrees", unit = "degrees")]
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData9 {
    #[simconnect(nameX = "PLANE LATITUDE", unit = "degrees")]
    pub lat: f64,
}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none")]
struct GpsData10 {
    #[simconnect(name = "PLANE LATITUDE", unitX = "degrees")]
    pub lat: f64,
}

fn main() {}
