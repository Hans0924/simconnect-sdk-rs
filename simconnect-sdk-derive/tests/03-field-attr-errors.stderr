error: expected attribute `#[simconnect(name = "...", unit = "...")]`
 --> tests/03-field-attr-errors.rs:7:5
  |
7 |     pub lat: f64,
  |     ^^^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:13:7
   |
13 |     #[simconnect]
   |       ^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:20:7
   |
20 |     #[simconnect()]
   |       ^^^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:27:7
   |
27 |     #[simconnect(name = "PLANE LATITUDE")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:34:7
   |
34 |     #[simconnect(unit = "degrees")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:41:7
   |
41 |     #[simconnect(name = "PLANE LATITUDE", name = "PLANE LATITUDE")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:48:7
   |
48 |     #[simconnect(unit = "degrees", unit = "degrees")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:55:7
   |
55 |     #[simconnect(name = "PLANE LATITUDE", unit = "degrees", unit = "degrees")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:62:7
   |
62 |     #[simconnect(nameX = "PLANE LATITUDE", unit = "degrees")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(name = "...", unit = "...")]`
  --> tests/03-field-attr-errors.rs:69:7
   |
69 |     #[simconnect(name = "PLANE LATITUDE", unitX = "degrees")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
